kernel/src/hil/sensors.rs

```diff
@@ -233,3 +234,23 @@ pub trait PressureClient {
     /// Returns the value in hPa.
     fn callback(&self, pressure: Result<u32, ErrorCode>);
 }
+
+/// Client for receiving NMEA sentances
+pub trait NmeaClient {
+    /// Called when a full NMEA sentance has been found, or if an I2C error
+    /// occurs.
+    fn callback(&self, buffer: &'static mut [u8], len: usize, status: Result<(), i2c::Error>);
+}
+
+/// A basic interface for a Nmea sentance reader
```

Comment:
This is missing a description of the various error cases that could happen here (e.g., how to report if the sensor is down / disconnected, if we can determine that, or whether the passed buffer is too small to fit the full sentence).